### PR TITLE
Add porting guide page to doxygen

### PR DESCRIPTION
**Pull request by andysun2015**
*Description of changes:*
Add porting guide page to doxygen.
- create docs/doxygen/porting.txt for porting guide page.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/shadow.c [hunk 1]
```diff
@@ -434,12 +434,11 @@ ShadowStatus_t Shadow_MatchTopic( const char * pTopic,
...
     uint16_t consumedTopicLength = 0U;
     ShadowStatus_t shadowStatus = SHADOW_SUCCESS;
+    uint16_t ThingNameLength = 0U;
 
     if( ( pTopic == NULL ) ||
         ( topicLength == 0U ) ||
-        ( pMessageType == NULL ) ||
-        ( pThingName == NULL ) ||
-        ( pThingNameLength == NULL ) )
+        ( pMessageType == NULL ) )
```
Comment (sarenameas) [inline source/shadow.c:441]: Why these changes?

Comment (andysun2015) [reply source/shadow.c]: remove from this commit.

File: source/shadow.c [hunk 2]
```diff
@@ -434,12 +434,11 @@ ShadowStatus_t Shadow_MatchTopic( const char * pTopic,
 {
     uint16_t consumedTopicLength = 0U;
     ShadowStatus_t shadowStatus = SHADOW_SUCCESS;
+    uint16_t ThingNameLength = 0U;
```
[the remark this answers is not in the document]
Comment (andysun2015) [reply source/shadow.c]: remove from this commit.

**Review by sarenameas — APPROVED** [review]

**Review by jainamt — APPROVED** [review]
